include/linker/rel-sections.ld

```diff
@@ -88,13 +97,23 @@
 	*(.rela.init_PRE_KERNEL*)
 	*(.rela.init_POST_KERNEL*)
 	*(.rela.init_APPLICATION*)
-	*(.rela._k_alert.*)
-	*(.rela._k_mem_pool.*)
-	*(.rela._k_timer.*)
+
+	*(.rela.devconfig.*)
+
+	*(.rela._k_*)
+
+	*(.rela._bt_settings.*)
+
+	*(.rela.log_*)
+
+	*(.rela._net_buf_pool.*)
+	*(.rela.net_*)
+
 	*(.rela._static_thread_data.*)
 
+	*(.rela.usb.*)
+
```

Comment:
Would it be dangerous to do *(.rela.*) ?

Reply:
I thought about that. However, couple platforms have explicit ".rela.plt" section with "*(.rela.plt)" (the same with ".rel.plt"). Maybe it would be fine since we are not doing relocation anyway. Let me try it and see.

Reply:
I tried collapsing all into one wildcard for rel and one for rela. However, native_posix fails to run unless there is a ".rel.plt" section. Then I tried having a ".rel.plt" section before the .rel.* one, and the native_posix binary runs fine. However, I haven't been able to figure out if the linker would honor the ordering. Since *(.rel.*) also matches *(.rel.plt.*), it really depends on the implementation of the linker.

Reply:
> However, I haven't been able to figure out if the linker would honor the ordering. Since (.rel.) also matches (.rel.plt.), it really depends on the implementation of the linker.

Luckily for us, it will honor the ordering. When an input section has multiple matches in the linker script it will always be placed in the first match.

Reply:
Good to know. I will update PR #10536 with wildcard ones.